### PR TITLE
Reg 19

### PR DESCRIPTION
**Pull request by arianaolson419**
https://olinjs.atlassian.net/browse/REG-19

### CODE REVIEW

**Comment by logandavis** [comment]
I don't think I'm fully on board with how you've split up your actions, props, and state for logging in vs. registering. I'd prefer more descriptive names than "USERNAME" vs. "NEW_USERNAME", and I'd prefer that the props and state refer to those attributes in the same way that the actions do. I think it might be good to spend some time thinking about state shape. Do the Login and Register usernames that you keep in state need to be different? Do they need different actions? If so, can you name those state fields descriptively?

**Review by logandavis — CHANGES_REQUESTED** [review]
I'd like more consistent names / organization for the state fields and the actions associated with the Register and Login components.

**Comment by logandavis** [comment]
Your commit messages are looking good! I notice that you have some commits with names like "Fix linter errors". Before you merge, I bet it'd be a good learning experience to use `git rebase -i` to clean up the commit history a little. That'd also help you fix the merge conflicts with master that you currently have.

File: src/components/RegisterNewUser.jsx
Comment (yunhsincynthiachen) [inline src/components/RegisterNewUser.jsx]: It doesn't really seem like we're doing anything really with "confirm password" field (ideally, it would do a check between this field and the new password field, and alert if not the same). Let's remove this for now, and tag this as future work, since it's not super crucial to implement for now.

File: src/components/LoginLocal.jsx
Comment (yunhsincynthiachen) [inline src/components/LoginLocal.jsx:1]: merge with master to get the style stuff in place

File: src/components/RegisterNewUser.jsx
Comment (yunhsincynthiachen) [inline src/components/RegisterNewUser.jsx]: merge with master to get the style stuff in place

File: src/actions/actions.jsx
Comment (yunhsincynthiachen) [inline src/actions/actions.jsx:49]: Is there any way we can have an alert error pop up when the user doesn't have an account/password is incorrect?

File: src/actions/actions.jsx
Comment (yunhsincynthiachen) [inline src/actions/actions.jsx]: maybe have "REGISTER", instead of "NEW", more specific to what your reducer is?

File: src/actions/actions.jsx
Comment (yunhsincynthiachen) [inline src/actions/actions.jsx]: I do agree with Logan, in that it would be a good idea to have keys that are more specific to what the component is.

**Review by yunhsincynthiachen — CHANGES_REQUESTED** [review]
Everything looks awesome, and really detailed. I have a couple changes, addressing some stuff Logan talked about, as well as removing unnecessary components. Also, master would be good to merge now.

**Comment by yunhsincynthiachen** [comment]
Good changes! Still  need to merge with master to make sure nothing will break, especially because Taylor's stuff is merged.

**Comment by Halliax** [comment]
This branch is no longer suffering from merge conflicts, but unfortunately login state doesn't persist. We've been sort of blocked on that problem for this afternoon

**Comment by billmwong** [comment]
We're putting the persisting login state bug in another ticket (REG-60), and going forward with this PR as-is.

- [x] Delete code that was used for testing so that this is production/master ready
- [x] Resolve conflicts with master again

**Comment by billmwong** [comment]
Resolve conflicts with master again

**Comment by yunhsincynthiachen** [comment]
Looks great! I tested the register and login functionalities, which work. Persist after redirect doesn't work, but @arianaolson419 is working on this in a different branch!

**Review by yunhsincynthiachen — APPROVED** [review]

**Review by robbie-sundstrom — APPROVED** [review]

**Pull request merged**
